var/spack/repos/builtin/packages/qscintilla/package.py

```diff
@@ -15,23 +15,37 @@ class Qscintilla(QMakePackage):
     homepage = "https://www.riverbankcomputing.com/software/qscintilla/intro"
     url      = "https://www.riverbankcomputing.com/static/Downloads/QScintilla/2.11.2/QScintilla_gpl-2.11.2.tar.gz"
 
+    # Directory structure is changed in latest release, logic is lost
+    version('2.12.0', sha256='a4cc9e7d2130ecfcdb18afb43b813ef122473f6f35deff747415fbc2fe0c60ed')
+
+    # Last standard release dates back to 2021/11/23
+    version('2.11.6', sha256='e7346057db47d2fb384467fafccfcb13aa0741373c5d593bc72b55b2f0dd20a7', preferred=True)
     version('2.11.2', sha256='029bdc476a069fda2cea3cd937ba19cc7fa614fb90578caef98ed703b658f4a1')
-    # Newer versions of Qscintilla won't build, so prefer the following version
-    version('2.10.2', sha256='14b31d20717eed95ea9bea4cd16e5e1b72cee7ebac647cba878e0f6db6a65ed0', preferred=True)
+    version('2.10.2', sha256='14b31d20717eed95ea9bea4cd16e5e1b72cee7ebac647cba878e0f6db6a65ed0')
 
     variant('designer', default=False, description="Enable pluging for Qt-Designer")
     variant('python', default=False, description="Build python bindings")
 
+    depends_on('qt+opengl',  when='+python')
     depends_on('qt')
     depends_on('py-pyqt5 +qsci_api', type=('build', 'run'),  when='+python ^qt@5')
     depends_on('py-pyqt4 +qsci_api', type=('build', 'run'),  when='+python ^qt@4')
     depends_on('python',   type=('build', 'run'),  when='+python')
+    # adter install inquires py-sip variant : so we need to have it
+    depends_on('py-sip',   type='build',  when='~python')
 
     extends('python', when='+python')
+    build_directory = 'Qt4Qt5'
+
+    def url_for_version(self, version):
+        if version >= Version('2.12.0'):
+            url = "https://www.riverbankcomputing.com/static/Downloads/QScintilla/{0}/QScintilla_src-{0}.tar.gz"
+        elif version <= Version('2.11.2'):
+            url = "https://www.riverbankcomputing.com/static/Downloads/QScintilla/{0}/QScintilla_gpl-{0}.tar.gz"
+        else:
+            url = "https://www.riverbankcomputing.com/static/Downloads/QScintilla/{0}/QScintilla-{0}.tar.gz"
```

Comment:
Since there are just a few versions, I would avoid coding `url_for_version` and just:
1. Update the class level url
2. Use `url=` kwargs in each version directive

Reply:
Just done

Reply:
There's one bit missing: 
- Update the class level url

If you use the latest url (and the package didn't change naming scheme in the meanwhile) we can extrapolate new versions with `spack versions` etc.

Reply:
OK, I've just done it.
This project has many other releases which I did not explicitly referred
[https://www.riverbankcomputing.com/software/qscintilla/download]
so don't you think an url_for_version would be more suitable ?

Reply:
@cessenat No, I don't think so. As far as I remember, but @adamjstewart can correct me if I'm wrong, the `url_for_version` doesn't currently enter into our extrapolation logic, so we are likely not to find new versions automatically with the `spack versions` command.

Also, if you think old versions are not needed anymore, we can now deprecate them.

Reply:
OK, I ignored that. Thanks for the explanation.
From the Web site of qscintilla there is no mention of deprecated version so I would not add this mention to even the 2.10.2.